scripts/check_macos.sh

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+
+# Checks that  README.md and the macos github action are up to date with macos_instructions.yml
```

Comment:
might want to consider some of the best practices in https://kvz.io/bash-best-practices.html
in particular the exit on error and warn on undefined variables (the `set -o` ones)

Reply:
Implemented the best practices.

Reply:
Still missing the `set -o` recommendations, would be helpful for debugging